pages/search/search.jsx

```diff
@@ -51,24 +50,97 @@ export default React.createClass({
       location[`query`] = { keyword: query };
     }
 
+    // note browserHistory.push() triggers component re-render
     browserHistory.push(location);
+  },
+  clearSearch() {
+    this.refs.searchInput.setState({value: ``}, () => {
+      this.updateBrowserHistory();
+    });
+  },
+  fetchData(params = {}, entries = [], apiPageIndex = 1) {
+    if (!params.search) {
+      // if no search keyword was passed reset search page back to its initial state
+      this.setState(this.getInitialState());
+      return;
+    }
+
+    params.page = apiPageIndex;
+
+    Service.entries
+      .get(params)
+      .then((data) => {
+        entries = entries.concat(data.results);
 
+        if (data.next) {
+          // there are more matched entries in the database we need to fetch
+          this.fetchData(params, entries, apiPageIndex+1);
```

Comment:
>  If we decide to make this change I suggest bumping up " the number of entries in each batch" from 24 to a bigger number. 

👍 as long as we don't overwhelm mobile users, loading 48 or more is fine by me. we could also pre-fetch as user scrolls down and nears the bottom. 

With page_size of 996, are we fetching that many _results_ or searching that many projects in total?  In 2017, we might cross 1000 entries to _search_. We would need to lift that limit b/c pace will increase as people anticipate new network site. I wouldn't, however, expect to cross 1000 search _results_ this year (for meaningful searches), so a cap on results is fine.